lstchain/mc/sensitivity.py

```diff
@@ -0,0 +1,184 @@
+import numpy as np
+import pandas as pd
+import astropy.units as u
+
+from eventio.simtel.simtelfile import SimTelFile
+
+from .plot_utils import sens_plot, sens_minimization_plot
+from .mc import rate, weight
+from lstchain.spectra.crab import crab_hegra
+from lstchain.spectra.proton import proton_bess
+
+
+def read_sim_par(source):
+    """
+    Read MC simulated parameters
+
+    Parameters
+    ---------
+    source: simtelarray file
+
+    Returns
+    ---------
+    par: `dict` with simulated parameters
+
+    """
+    emin, emax = source.mc_run_headers[0]['E_range'] * u.TeV
+    sp_idx = source.mc_run_headers[0]['spectral_index']
+    n_showers = source.mc_run_headers[0]['num_showers']
+    n_use = source.mc_run_headers[0]['num_use']
+    sim_ev= n_showers * n_use
+    max_impact = source.mc_run_headers[0]['core_range'][1] * u.m 
+    area_sim = np.pi * np.power(max_impact,2)
+    cone = source.mc_run_headers[0]['viewcone'][1] * u.deg
+
+    par_var = [emin, emax, sp_idx, sim_ev, area_sim, cone]
+    par_dic = ['emin', 'emax', 'sp_idx', 'sim_ev', 'area_sim', 'cone']
+    par = dict(zip(par_dic, par_var))
+
+    return par
+
+def process_mc(simtelfile, dl2_file):
+    """
+    Process the MC simulated and triggered to extract the relevant
+    parameters to compute the sensitivity
+
+    Paramenters
+    ---------
+    simtel: simtelarray file
+    dl2_file: `pandas.DataFrame` dl2 parameters
+
+    Returns
+    ---------
+    gammaness: `numpy.ndarray` 
+    theta2:    `numpy.ndarray` 
+    e_trig:    `numpy.ndarray` triggered energies
+    n_trig:    `int` number of triggered events
+    mc_par:    `dict` with simulated parameters
+
+    """
+    source = SimTelFile(simtelfile)
+    sim_par = read_sim_par(source)
+    events = pd.read_hdf(dl2_file)
+
+    e_trig = 10**events.mc_energy * u.GeV
+    n_trig = e_trig.shape[0]
+    gammaness = events.gammaness
+    theta2 = (events.src_x - events.src_x_rec)**2 + \
```

Comment:
shouldn't this be angular separation?

Reply:
what do you mean? 
we define theta2 (for gammas) as `(events.src_x - events.src_x_rec)**2 + (events.src_y - events.src_y_rec)**2 * u.deg**2`

Reply:
`src_x` and alikes are angles right?
and theta is therefore the angular separation between these directions?
Shouldn't it be calculated following https://en.wikipedia.org/wiki/Great-circle_distance ?
I think this is what is used in astropy and ctaplot.

Reply:
is this really significant for the small angle approximation we are working in? We can of course improve it later if that is ok

Reply:
I don't know how significant is the difference